go/tests/endpoint/endpoint.go

```diff
@@ -54,6 +54,20 @@ type fn func()
 
 var cleanup fn = func() {}
 
+var cwd, err_cwd = os.Getwd()
+var bin, err_bin = os.Executable()
+
+func init() {
+	if err_cwd == nil && err_bin == nil {
+		bindir := filepath.Dir((bin))
+		if bindir != cwd {
+			os.Chdir(bindir)
+			cwd = bindir
+			Say("Executing out of '%s'", cwd)
```

Comment:
also recommend surfacing this Say call to maximize clarity in logs. currently only runs if cwdErr or binErr, but i think it's most useful if it's always there.

otherwise keep inside the block but make it more expressive. something to indicate cwd isn't binDir and that's going to be changed at time of test.